### PR TITLE
vicodyn: add `X-Error-Generated-By` to replies with an error

### PR DESCRIPTION
**Pull request by shaitan**
Depends on https://github.com/3Hren/cocaine-plugins/pull/201

### CODE REVIEW

File: vicodyn/src/vicodyn/proxy.cpp [hunk 1]
```diff
@@ -22,6 +22,12 @@
 
 #include <cocaine/traits/map.hpp>
 
+namespace {
+
+static const cocaine::hpack::header_t VICODYN_ERROR_HEADER{"X-Error-Generated-By", "Vicodyn"};
```
Comment (antmat) [inline vicodyn/src/vicodyn/proxy.cpp]: style: uppercase is reserved for macro. constants are declared the same way as vars

**Review by antmat — COMMENTED** [review]
We already have a mechanism of error_codes. In proxy you can check for error code and return the right http code for examle. I recommend you to fix the codes (e.g. in maximum retries reached and similar), possibly creating a new error_category for vicodyn and use this mechanism to generate proper response.

File: vicodyn/src/vicodyn/proxy.cpp (hunk 1 above)
Comment (shaitan) [reply vicodyn/src/vicodyn/proxy.cpp]: done

**Comment by 3Hren** [comment]
Please rebase changes and bump version.

**Comment by shaitan** [comment]
done

**Comment by shaitan** [comment]
thnx

**Pull request merged**